Stripe/STPCustomer.m

```diff
@@ -58,29 +58,34 @@ - (NSString *)description {
 
 #pragma mark - STPAPIResponseDecodable
 
-+ (NSArray *)requiredFields {
-    return @[@"id"];
-}
-
 + (nullable instancetype)decodedObjectFromAPIResponse:(nullable NSDictionary *)response {
-    NSDictionary *dict = [response stp_dictionaryByRemovingNullsValidatingRequiredFields:[self requiredFields]];
+    NSDictionary *dict = [response stp_dictionaryByRemovingNulls];
     if (!dict) {
         return nil;
     }
 
+    // required fields
+    NSString *stripeId = [dict stp_stringForKey:@"id"];
+    if (!stripeId) {
+        return nil;
+    }
+
     STPCustomer *customer = [[self class] new];
-    customer.stripeID = dict[@"id"];
-    if ([dict[@"shipping"] isKindOfClass:[NSDictionary class]]) {
-        NSDictionary *shippingDict = dict[@"shipping"];
+    customer.stripeID = stripeId;
+    NSDictionary *shippingDict = [dict stp_dictionaryForKey:@"shipping"];
+    if (shippingDict) {
         STPAddress *shipping = [STPAddress new];
-        shipping.name = shippingDict[@"name"];
-        shipping.phone = shippingDict[@"phone"];
-        shipping.line1 = shippingDict[@"address"][@"line1"];
-        shipping.line2 = shippingDict[@"address"][@"line2"];
-        shipping.city = shippingDict[@"address"][@"city"];
-        shipping.state = shippingDict[@"address"][@"state"];
-        shipping.postalCode = shippingDict[@"address"][@"postal_code"];
-        shipping.country = shippingDict[@"address"][@"country"];
+        shipping.name = [shippingDict stp_stringForKey:@"name"];
+        shipping.phone = [shippingDict stp_stringForKey:@"phone"];
+        NSDictionary *addressDict = [shippingDict stp_dictionaryForKey:@"address"];
+        if (addressDict) {
```

Comment:
WDYT about refactoring/reordering this method to use `+[STPAddress decodedObjectFromAPIResponse:]` to extract the address lines? It's not totally straightforward because you have to handle the nil address dict case and still put name/phone into the shipping address, but...

Reply:
ah, that's a good idea 👍